### PR TITLE
fix(release): improve process and dockerfile again

### PR DESCRIPTION
**Pull request by bodinsamuel**
- Make release only manual for the moment, docker is taking a huge amount of time to compile because we need multiple platforms and with it's not ideal to release that much for the moment.

- Add a `dry run` option to release 

- Dockerfile: cache dependencies more aggressively by leveraging a 4th layer (yes), copying the package.json and remove version and useless packages. This should improve build time by a lot and size maybe. And help release because when we release we increment the `version` so it was never using the cache.

## Test

```sh
yarn docker:build
```

The first build will be slow and next should be instant.
Then if you modify the version in package.json it should still be instant.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -14,6 +16,12 @@ jobs:
     runs-on: ubuntu-latest
     name: Release
     if: (github.event.workflow_run.conclusion == 'success')
+    env:
+      GH_TOKEN: ${{ secrets.GH_TOKEN }}
+      GIT_AUTHOR_NAME: ${{ secrets.GH_USER_NAME }}
+      GIT_AUTHOR_EMAIL:	${{ secrets.GH_USER_EMAIL }}
+      GIT_COMMITTER_NAME:	${{ secrets.GH_USER_NAME }}
```
Comment (millotp) [inline .github/workflows/release.yml:23]:
```suggestion
      GIT_COMMITTER_NAME: ${{ secrets.GH_USER_NAME }}
```

**Review by millotp — APPROVED** [review]
The new dockerfile seems to work great ! However with `buildx` it's not possible to test the image locally, at least I can't.
Do you think it's possible to have a `build-local.sh` script that build only for the current platform and use the `--load` option to push to local registry ? That way it's possible to use `test_image.sh` locally

**Comment by bodinsamuel** [comment]
Yep I don't know how to test it with `buildx` it's weird.
I have built manually to test. why everything has to be complicated 😞

**Review by millotp — APPROVED** [review]
As long as it's possible to test locally that's cool, not a priority right now but could be documented

**Pull request merged**